js/main.js

```diff
@@ -1,45 +1,491 @@
+function cpuFire(board) {
+  let randomDivIdx;
+
+  if (lastHitRow !== null && lastHitColumn !== null) {
+    suitableIndex = parseInt(getSuitableIndex(board));
+  }
+  if (
+    suitableIndex === null ||
+    isNaN(suitableIndex) ||
+    LOCATIONS.has(suitableIndex)
+  ) {
+    do {
+      randomDivIdx = getRandomNum(0, 99);
+    } while (LOCATIONS.has(randomDivIdx));
+  } else randomDivIdx = suitableIndex;
+
+  LOCATIONS.add(randomDivIdx);
+  let randomDiv = PLAYER_DIVS[randomDivIdx];
+  let rows = parseInt(randomDivIdx.toString().charAt(0));
+  let columns = parseInt(randomDivIdx.toString().charAt(1));
+
+  if (isNaN(columns)) {
+    columns = rows;
+    rows = 0;
+  }
+  let currentShip;
+
+  if (PLAYER_DIVS[randomDivIdx].classList.contains("player-ship")) {
+    currentShip = findCorrectShip(PLAYER_BOARD, randomDiv);
+    if (!PLAYER_DIVS[randomDivIdx].classList.contains("hit-ship"))
+      currentShip.health--;
+    lastHitColumn = columns;
+    lastHitRow = rows;
+    checkIfSunk(PLAYER_BOARD, currentShip, randomDiv);
+    PLAYER_DIVS[randomDivIdx].classList.add("hit-ship");
+    PLAYER_BOARD[rows][columns] = 2;
+  }
+  if (PLAYER_BOARD[rows][columns] === 0) {
+    randomDiv.classList.add("miss");
+    PLAYER_BOARD[rows][columns] = 3;
+  }
+  turn = 1;
+  activeTurn();
+}
+
+//Finding if there is an index around the last hit div that can be fired at again
+function getSuitableIndex(board) {
+  if (
+    lastHitColumn + 1 !== 10 &&
+    board[lastHitRow][lastHitColumn + 1] !== 2 &&
+    board[lastHitRow][lastHitColumn + 1] !== 3
+  ) {
+    return `${lastHitRow}${lastHitColumn + 1}`;
+  } else if (
+    lastHitRow + 1 !== 10 &&
+    board[lastHitRow + 1][lastHitColumn] !== 2 &&
+    board[lastHitRow + 1][lastHitColumn] !== 3
+  ) {
+    return `${lastHitRow + 1}${lastHitColumn}`;
+  } else if (
+    lastHitColumn - 1 !== -1 &&
+    board[lastHitRow][lastHitColumn - 1] !== 2 &&
+    board[lastHitRow][lastHitColumn - 1] !== 3
+  ) {
+    return `${lastHitRow}${lastHitColumn - 1}`;
+  } else if (
+    lastHitRow - 1 !== -1 &&
+    board[lastHitRow - 1][lastHitColumn] !== 2 &&
+    board[lastHitRow - 1][lastHitColumn] !== 3
+  ) {
+    return `${lastHitRow - 1}${lastHitColumn}`;
+  } else return null;
+}
+
+//Get the correct ship object from the clicked div
+function findCorrectShip(board, div) {
+  let correctShip;
+  Object.values(board === PLAYER_BOARD ? playerShips : cpuShips).forEach(
+    (obj) => {
+      if (obj.key === div.id) correctShip = obj;
+    }
+  );
+  return correctShip;
+}
+
+//Checking if the ship that was hit has been sunk
+function checkIfSunk(board, ship, div) {
+  let currentPlayer = board === PLAYER_BOARD ? PLAYER_DIVS : CPU_DIVS;
+  if (ship.health === 0 && [...div.classList].includes("horizontal")) {
+    for (let i = 0; i < ship.length; i++) {
+      currentPlayer[parseInt(ship.firstIndex) + i].classList.add("sunken-ship");
+    }
+    if (board === PLAYER_BOARD) {
+      suitableIndex = null;
```

Comment:
This (and when you do it again below) can be simplified to something like `suitableIndex = lastHitRow = lastHitColumn = null`